### ISSUE TITLE
Perf sucks

### ISSUE THREAD

**Issue by WasabiFan** [kyranstar/Hunt-The-Wumpus#38]
Make our code more fasterer.

**Comment by kyranstar** [comment]
Our biggest slowdowns are the particle systems, I just measured. They take almost a hundred times as much time as the rest of the render code. Perhaps if it runs slowly on the demo computer we can just disable the particle systems? I'll write a command for it now.

**Comment by WasabiFan** [comment]
OK; sounds good. I'm currently working on making the game reset correctly.

**Comment by WasabiFan** [comment]
We can also try to make some of that code faster -- maybe put some of it in an `unsafe` block and regress to the pointer age?

**Comment by nikbyk** [comment]
@WasabiFan what did you do after you rolled back to 10074 to get start and action center back?

**Comment by WasabiFan** [comment]
Uhhhhh... I gave up? I think I'm just going to try to upgrade to 10122 again, and maybe revert again. Why, did you hit the same issue?

**Comment by nikbyk** [comment]
yep did u run sfc?

**Comment by WasabiFan** [comment]
I ran it and it said that it fixed errors... but the problem didn't go away.

**Comment by nikbyk** [comment]
okay, ill try and see

**Comment by nikbyk** [comment]
wait, fixed vs errors?

**Comment by nikbyk** [comment]
we should have just done that on 10122 then

**Comment by WasabiFan** [comment]
It said that it had found corrupted files and repaired them, but after restarting I still couldn't open the Start menu

**Comment by WasabiFan** [comment]
> wait, fixed vs errors?

No, not the VS errors, sadly

**Comment by nikbyk** [comment]
ah, damn. even in 10074?

**Comment by WasabiFan** [comment]
No; it was fine in 10074 from the start. Just not on 10122.

P.S. we probably shouldn't be discussing this in this issue ;)

**Comment by nikbyk** [comment]
lol, yea ur right

**Comment by nikbyk** [comment]
sorry for continuing here, but I can't get back to 10122, or get the start menu working

**Comment by WasabiFan** [comment]
Why not? What's the issue?

**Comment by nikbyk** [comment]
Its not in update. Doesnt show up

**Comment by WasabiFan** [comment]
Make sure that you're in the fast ring, restart, and try again?

**Comment by nikbyk** [comment]
i did

**Comment by WasabiFan** [comment]
Hmmm... I don't know then -- that's what I did

**Comment by nikbyk** [comment]
you're back on 10122?

**Comment by nikbyk** [comment]
ill keep trying i guess

**Comment by WasabiFan** [comment]
I'm 64% of the way there

**Comment by nikbyk** [comment]
ok, lemme know if your start menu comes back

**Comment by nikbyk** [comment]
device is up to date

**Comment by WasabiFan** [comment]
If it says that you're on build *74 and it still doesn't show any available updates, I really have no idea what's going on... maybe try to restore to the restore point that it creates just before the upgrade from *74 to *122?

**Comment by nikbyk** [comment]
how would I do that?

**Comment by WasabiFan** [comment]
Settings > Update > Recovery, I think. If there isn't an option to restore from a restore point there, select "advanced startup" and when it reboots click "troubleshoot > advanced > restore point.

P.S. It's really sad that I know that from memory

**Comment by nikbyk** [comment]
Lol, it kinda is


Sent by Outlook for Android





On Thu, May 21, 2015 at 9:00 PM -0700, "Wasabi Fan" <notifications@github.com> wrote:
Settings > Update > Recovery, I think. If there isn't an option to restore from a restore point there, select "advanced startup" and when it reboots click "troubleshoot > advanced > restore point.

P.S. It's really sad that I know that from memory

---

**Comment by nikbyk** [comment]
I don't see a restore point option

**Comment by WasabiFan** [comment]
In which menu?

**Comment by nikbyk** [comment]
Troubleshoot. Just has uefi and startup settings

**Comment by nikbyk** [comment]
I'll just install from I so I giess

**Comment by WasabiFan** [comment]
Huh... IDK then. My re-upgrade seems to have worked (?)

**Comment by WasabiFan** [comment]
I'm extremely confused. After re-upgrading, this error seems to have gone away.. I can now build and run our game.

**Comment by nikbyk** [comment]
What? How?


Sent by Outlook for Android





On Thu, May 21, 2015 at 9:39 PM -0700, "Wasabi Fan" <notifications@github.com> wrote:
I'm extremely confused. After re-upgrading, this error seems to have gone away.. I can now build and run our game.

---

**Comment by WasabiFan** [comment]
I don't know... but I can't repro the issue anymore

**Comment by nikbyk** [comment]
Well, I'll try too once I get to 10122


Sent by Outlook for Android





On Thu, May 21, 2015 at 9:41 PM -0700, "Wasabi Fan" <notifications@github.com> wrote:
I don't know... but I can't repro the issue anymore

---

**Comment by WasabiFan** [comment]
K; I'd like to see what happens.

**Comment by nikbyk** [comment]
Me too. Assuming I can upgrade after reinstalling 10074


Sent by Outlook for Android





On Thu, May 21, 2015 at 9:42 PM -0700, "Wasabi Fan" <notifications@github.com> wrote:
K; I'd like to see what happens.

---

**Comment by WasabiFan** [comment]
Before you upgrade from *74, can you install VS, clone the repo, and run a build? I think it could be related to being installed during the upgrade.

**Comment by nikbyk** [comment]
Will do


Sent by Outlook for Android





On Thu, May 21, 2015 at 9:44 PM -0700, "Wasabi Fan" <notifications@github.com> wrote:
Before you upgrade from *74, can you install VS, clone the repo, and run a build? I think it could be related to being installed during the upgrade.

---

**Comment by nikbyk** [comment]
Runs on 10074

**Comment by nikbyk** [comment]
If windows update works, should be moving to 10122 soon

**Comment by WasabiFan** [comment]
Yup; as long as you know that it runs on *74

**Comment by nikbyk** [comment]
Some updates currently stuck at 0%. No 10122 in sight yet.

**Comment by nikbyk** [comment]
@WasabiFan System Firmware Update - ‎5/‎14/‎2015 - Error 0x80070002

**Comment by nikbyk** [comment]
All updates installed, but no 10122 in windows update :(

**Comment by WasabiFan** [comment]
Are you sure that you are in the fast ring and connected to Wi-Fi?

**Comment by nikbyk** [comment]
Yep. I'll give it until tomorrow and see if it works


Sent by Outlook for Android





On Thu, May 21, 2015 at 10:55 PM -0700, "Wasabi Fan" <notifications@github.com> wrote:
Are you sure that you are in the fast ring and connected to Wi-Fi?

---

**Comment by WasabiFan** [comment]
http://community.monogame.net/t/windows-10-insider-preview-10122-appears-to-break-content-pipeline/2479

**Comment by nikbyk** [comment]
But ur errors went away?


Sent by Outlook for Android





On Fri, May 22, 2015 at 10:27 AM -0700, "Wasabi Fan" <notifications@github.com> wrote:
http://community.monogame.net/t/windows-10-insider-preview-10122-appears-to-break-content-pipeline/2479

---

**Comment by WasabiFan** [comment]
Yep... I'm thinking it's an issue with corrupted files on upgrade. After I reinstalled, maybe it fixed some file (?)

That's why I want to see what your machine does.

**Comment by nikbyk** [comment]
Well, I still can't get to 10122


Sent by Outlook for Android





On Fri, May 22, 2015 at 10:43 AM -0700, "Wasabi Fan" <notifications@github.com> wrote:
Yep... I'm thinking it's an issue with corrupted files on upgrade. After I reinstalled, maybe it fixed some file (?)

That's why I want to see what your machine does.

---

**Comment by nikbyk** [comment]
I still cant get to 10122: Your device is up to date
Any ideas?

**Comment by WasabiFan** [comment]
I'm out of ideas, honestly -- maybe look at your WU history? Other than that, I can't think of anything at the moment. I'll take a look around on my device and see if I get any ideas.

**Comment by nikbyk** [comment]
Should I uninstall updates or last build?

**Comment by WasabiFan** [comment]
You could try it, but I wouldn't recommend it -- rolling back is not something that they test very much.

**Comment by nikbyk** [comment]
ok

**Comment by nikbyk** [comment]
I give up, I'll install from iso

**Comment by WasabiFan** [comment]
Do they really have an ISO of 10122 yet?

**Comment by nikbyk** [comment]
nope, people make them from esd files

**Comment by WasabiFan** [comment]
Oh great that sounds safe :confused:

**Comment by nikbyk** [comment]
yep, but people seem to have used them before

**Comment by nikbyk** [comment]
no errors on 10122

**Comment by WasabiFan** [comment]
Hmmm... I think they might have released a patch, but haven't gotten a chance to check yet.

**Comment by nikbyk** [comment]
There was an update to fix amd driver issues


Sent by Outlook for Android





On Fri, May 22, 2015 at 1:26 PM -0700, "Wasabi Fan" <notifications@github.com> wrote:
Hmmm... I think they might have released a patch, but haven't gotten a chance to check yet.

---

**Comment by nikbyk** [comment]
Also, about the game, after a pull no trivia or end game shows up. Opacity of those is 0

**Comment by WasabiFan** [comment]
@nikbyk Try now

**Comment by nikbyk** [comment]
Great, works now


Sent by Outlook for Android





On Fri, May 22, 2015 at 2:38 PM -0700, "Wasabi Fan" <notifications@github.com> wrote:
@nikbyk Try now

---
